include/aspect/particle/property/interface.h

```diff
@@ -70,6 +70,11 @@ namespace aspect
       class Interface
       {
         public:
+          /**
+           * Destructor. Made virtual so that derived classes can be created
+           * and destroyed through pointers to the base class.
+           */
+          virtual ~Interface () {}
```

Comment:
We're generally not used to putting the implementation of functions into the class declaration, even if the function is empty. Please move it to the .cc file and then merge yourself.